From 7b465d3f1e11811b4b3fd873514a939e249f247f Mon Sep 17 00:00:00 2001
From: Nithya <nithya.mani@caviumnetworks.com>
Date: Mon, 9 May 2016 12:03:17 -0700
Subject: [PATCH 0414/1239] Fix the 81xx config for crash during sata init

SATA and USB devices are enumerated using PCI
and required configuration was not enabled for
81xx. Fixed the config.

Signed-off-by: Nithya <nithya.mani@caviumnetworks.com>
---
 configs/thunderx_81xx_defconfig | 3 +--
 include/configs/thunderx_81xx.h | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/configs/thunderx_81xx_defconfig b/configs/thunderx_81xx_defconfig
index 90c6001bcf..dc3694408a 100644
--- a/configs/thunderx_81xx_defconfig
+++ b/configs/thunderx_81xx_defconfig
@@ -1,6 +1,5 @@
 CONFIG_ARM=y
 CONFIG_TARGET_THUNDERX_81XX=y
-CONFIG_DM_SERIAL=y
 CONFIG_DEFAULT_DEVICE_TREE="thunder-81xx"
 CONFIG_SYS_EXTRA_OPTIONS="ARM64"
 CONFIG_HUSH_PARSER=y
@@ -21,7 +20,7 @@ CONFIG_CMD_SNTP=y
 CONFIG_CMD_DNS=y
 CONFIG_CMD_LINK_LOCAL=y
 CONFIG_NET_RANDOM_ETHADDR=y
-CONFIG_DM=y
+CONFIG_PCI_THUNDERX_ECAM=y
 CONFIG_DEBUG_UART=y
 CONFIG_DEBUG_UART_PL011=y
 CONFIG_DEBUG_UART_BASE=0x87e028000000
diff --git a/include/configs/thunderx_81xx.h b/include/configs/thunderx_81xx.h
index 150c6d43b4..ad488d200d 100644
--- a/include/configs/thunderx_81xx.h
+++ b/include/configs/thunderx_81xx.h
@@ -185,8 +185,7 @@
 /* AHCI support Definitions */
 #ifdef CONFIG_CMD_SATA
   #define CONFIG_SATA_AHCI
-  #define CONFIG_SYS_SATA_MAX_DEVICE	32
-  #define CONFIG_SATA_BASE_ADDR		0x810000000000
+  #define CONFIG_SYS_SATA_MAX_DEVICE	2
   #define CONFIG_LBA48
   #define CONFIG_LIBATA
   #define CONFIG_SYS_64BIT_LBA
-- 
2.29.0

